### PR TITLE
#45 Add TextInputLayout.bindFormField

### PR DESCRIPTION
**Pull request by ExNDY**
- add fields-material module for android platform with binding for textinputlayout
- include update same libs

### CODE REVIEW

File: fields-flow/build.gradle.kts [hunk 1]
```diff
@@ -10,6 +10,8 @@ plugins {
 }
 
 dependencies {
+    androidMainApi(libs.androidXLifecycle)
+    androidMainApi(libs.lifecycleKtx)
```
Comment (Alex009) [inline fields-flow/build.gradle.kts:14]: it's already available as api in `fieldsCore`

File: fields-livedata/build.gradle.kts [hunk 2]
```diff
@@ -10,6 +10,8 @@ plugins {
 }
 
 dependencies {
+    androidMainApi(libs.androidXLifecycle)
+    androidMainApi(libs.lifecycleKtx)
```
Comment (Alex009) [inline fields-livedata/build.gradle.kts:14]: already available from fieldsCore

File: sample/android-app/build.gradle.kts [hunk 3]
```diff
@@ -20,9 +20,16 @@ android {
 }
 
 dependencies {
+    configurations {
+        all {
+            exclude("androidx.lifecycle","lifecycle-viewmodel-ktx")
+        }
+    }
```
Comment (Alex009) [inline sample/android-app/build.gradle.kts:27]: why?

File: sample/mpp-library/src/commonMain/kotlin/com/icerockdev/library/LoginViewModel.kt [hunk 4]
```diff
@@ -11,6 +11,7 @@ import dev.icerock.moko.fields.core.validations.ValidationResult
 import dev.icerock.moko.fields.core.validations.matchRegex
 import dev.icerock.moko.fields.core.validations.minLength
 import dev.icerock.moko.fields.core.validations.notBlank
+import dev.icerock.moko.fields.livedata.BaseLiveDataFormField
```
Comment (Alex009) [inline sample/mpp-library/src/commonMain/kotlin/com/icerockdev/library/LoginViewModel.kt:14]: why here this type?

**Review by Alex009 — CHANGES_REQUESTED** [review]

File: sample/android-app/build.gradle.kts (hunk 3 above)
Comment (ExNDY) [reply sample/android-app/build.gradle.kts]: Sample has error on build:
`Duplicate class androidx.lifecycle.ViewModelLazy found in modules lifecycle-viewmodel-2.5.0-runtime (androidx.lifecycle:lifecycle-viewmodel:2.5.0) and lifecycle-viewmodel-ktx-2.2.0-runtime (androidx.lifecycle:lifecycle-viewmodel-ktx:2.2.0)`

An exception has been added to resolve the conflict two libs. But, now im add lifecycle-viewmodel-ktx lib in sample and concretize version and problem was solution.

File: sample/mpp-library/src/commonMain/kotlin/com/icerockdev/library/LoginViewModel.kt (hunk 4 above)
Comment (ExNDY) [reply sample/mpp-library/src/commonMain/kotlin/com/icerockdev/library/LoginViewModel.kt]: So sorry, My mistake in tests performance livedata and fixes after core changes. Removed

File: fields-livedata/build.gradle.kts (hunk 2 above)
Comment (ExNDY) [reply fields-livedata/build.gradle.kts]: My mistake, removed

File: fields-flow/build.gradle.kts (hunk 1 above)
Comment (ExNDY) [reply fields-flow/build.gradle.kts]: My mistake, removed

**Review by Alex009 — APPROVED** [review]

**Pull request merged**